addons/point_of_sale/models/pos_session.py

```diff
@@ -308,19 +309,35 @@ def _validate_session(self):
             # Users without any accounting rights won't be able to create the journal entry. If this
             # case, switch to sudo for creation and posting.
             sudo = False
-            if (
-                not self.env['account.move'].check_access_rights('create', raise_exception=False)
-                and self.user_has_groups('point_of_sale.group_pos_user')
-            ):
-                sudo = True
-                self.sudo().with_company(self.company_id)._create_account_move()
-            else:
-                self.with_company(self.company_id)._create_account_move()
-            if self.move_id.line_ids:
-                # Set the uninvoiced orders' state to 'done'
-                self.env['pos.order'].search([('session_id', '=', self.id), ('state', '=', 'paid')]).write({'state': 'done'})
-            else:
-                self.move_id.unlink()
+            try:
+                if (
+                    not self.env['account.move'].check_access_rights('create', raise_exception=False)
+                    and self.user_has_groups('point_of_sale.group_pos_user')
+                ):
+                    sudo = True
+                    self.sudo().with_company(self.company_id)._create_account_move(balancing_account, amount_to_balance)
+                else:
+                    self.with_company(self.company_id)._create_account_move(balancing_account, amount_to_balance)
+                if self.move_id.line_ids:
+                    # Set the uninvoiced orders' state to 'done'
+                    self.env['pos.order'].search([('session_id', '=', self.id), ('state', '=', 'paid')]).write({'state': 'done'})
+                else:
+                    self.move_id.unlink()
+
+            except ImbalanceJournalEntryError as error:
+                # Creating the account move is just part of a big database transaction
+                # when closing a session. There are other database changes that will happen
+                # before attempting to create the account move, such as, creating the picking
+                # records.
+                # We don't, however, want them to be committed when the account move creation
+                # failed; therefore, we need to roll back this transaction before showing the
+                # close session wizard.
+                self.env.cr.rollback()
+                # When closing a session, single account move is checked which resulted to
+                # `ImbalanceJournalEntryError`. Thus, there is only single value in imbalance_amounts
+                # which is what we need.
+                return self._close_session_action(error.imbalance_amounts[0])
```

Comment:
You are masking some errors if there are multiple `imbalance_amounts` here. You should still raise an error if it is the case, or manage it correctly.